airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/streams.py

```diff
@@ -0,0 +1,306 @@
+        """
+        next_params = super().next_page_token(response)
+        if not next_params:
+            return None
+
+        current_url = urlparse(response.request.url)
+        current_params = parse_qs(current_url.query)
+
+        # check if cursor value was changed
+        if current_params[self.filter_param] != next_params[self.filter_param]:
+            return next_params
+
+        return None
+
+
+class PhoneNumbers(ZendeskTalkStream):
+    """Phone Numbers
+    Docs: https://developer.zendesk.com/api-reference/voice/talk-api/phone_numbers/#list-phone-numbers
+    """
+
+    data_field = "phone_numbers"
+
+    def path(self, **kwargs) -> str:
+        return "/phone_numbers"
+
+
+class Addresses(ZendeskTalkStream):
+    """Addresses
+    Docs: https://developer.zendesk.com/api-reference/voice/talk-api/addresses/#list-addresses
+    """
+
+    data_field = "addresses"
+
+    def path(self, **kwargs) -> str:
+        return "/addresses"
+
+
+class GreetingCategories(ZendeskTalkStream):
+    """Greeting Categories
+    Docs: https://developer.zendesk.com/rest_api/docs/voice-api/greetings#list-greeting-categories
+    """
+
+    data_field = "greeting_categories"
+
+    def path(self, **kwargs) -> str:
+        return "/greeting_categories"
+
+
+class Greetings(ZendeskTalkStream):
+    """Greetings
+    Docs: https://developer.zendesk.com/rest_api/docs/voice-api/greetings#list-greetings
+    """
+
+    data_field = "greetings"
+
+    def path(self, **kwargs) -> str:
+        return "/greetings"
+
+
+class IVRs(ZendeskTalkStream):
+    """IVRs
+    Docs: https://developer.zendesk.com/rest_api/docs/voice-api/ivrs#list-ivrs
+    """
+
+    name = "ivrs"
+    data_field = "ivrs"
+    use_cache = True
+    cache_filename = "ivrs.yml"
+
+    def path(self, **kwargs) -> str:
+        return "/ivr.json"
+
+
+class IVRMenus(IVRs):
+    """IVR Menus
+    Docs: https://developer.zendesk.com/rest_api/docs/voice-api/ivrs#list-ivrs
+    """
+
+    name = "ivr_menus"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """ Simply parse json and iterates over root object"""
+        ivrs = super().parse_response(response=response, **kwargs)
+        for ivr in ivrs:
+            for menu in ivr["menus"]:
+                yield {"ivr_id": ivr["id"], **menu}
+
+
+class IVRRoutes(IVRs):
+    """IVR Routes
+    Docs: https://developer.zendesk.com/rest_api/docs/voice-api/ivr_routes#list-ivr-routes
+    """
+
+    name = "ivr_routes"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """ Simply parse json and iterates over root object"""
+        ivrs = super().parse_response(response=response, **kwargs)
+        for ivr in ivrs:
+            for menu in ivr["menus"]:
+                for route in ivr["menus"]:
+                    yield {"ivr_id": ivr["id"], "ivr_menu_id": menu["id"], **route}
+
+
+class AccountOverview(ZendeskTalkStream):
```

Comment:
it feels like to make this stream useful, we might want to add the current timestamp as an extra field and make it the primary key. This will allow us to make it incremental, and will allow the user to deduplicate based on the latest timestamp. WDYT?

Reply:
another great idea, will do